File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] newPath = path.split("/");
        Stack<String> ans = new Stack<>();
        String n = "";
        for(String i : newPath){
            if(i.equals("..") && !ans.isEmpty()) ans.pop();
            else if(i.equals("..") && ans.isEmpty());
            else if(i.equals(".") ||  i.equals("")) ;
            else ans.push("/" + i);
        }
        if(ans.isEmpty()) ans.push("/");
        for(String i : ans){
            n += i;
        }
        return n;
    }
}

File: 0071-simplify-path/0071-simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    let strayed = path.split("/")
    let result = []
    strayed.forEach((item) => {
        if(item === ".."){
            result.pop()
        }else if(item !== "." && item !== ""){
            result.push(`/${item}`)
        }
    })
    if(result.length === 0) result.push("/")
    return result.join("")
};